_source/_posts/2022-08-08-react-redux.md

```diff
@@ -0,0 +1,123 @@
+
+1. First, run:
+```bash
+npx create-react-app <YOUR_APP_NAME> --template redux-typescript
+```
+
+2. Then we'll add the Redux core by running:
+```bash
+npm add redux
+```
+
+3. Redux provides its own types, but we'll want to add our `react-redux` types since we're using Typescript:
+```bash
+npm add -D @types/react-redux
+```
+
+**NOTE:** Redux also recommends you install the [Chrome React DevTools Extension](https://chrome.google.com/webstore/detail/react-developer-tools/fmkadmapgofadopljbjfkapdkoienihi?hl=en).
+
+4. We'll also need routing (and routing types) for our app:
+```bash
+npm install react-router-dom@5
+npm add -D @types/react-router-dom
+```
+
+5. To start your app, run:
+```bash
+npm start
+```
+
+## How does Redux work?
+
+To update state using Redux, an action is dispatched and the store then uses the root reducer to calculate a new state as compared to the old state, which then notifies the proper subscribers of the update so that the UI can properly be updated.
+
+A typical redux workflow goes like this:
+
+1. xyz
+2. abc
+
+### Terms
+
+####[Action](https://redux.js.org/tutorials/essentials/part-1-overview-concepts#reducers)
+
+An Action is an JS object with a type and a payload. It describes an actionable interaction within your application, and typically is named something like `createUser` or `addToDo`.
+
+####[Action Creators](https://redux.js.org/tutorials/essentials/part-1-overview-concepts#reducers)
+
+Action Creators can be used to dynamically create actions. NOT IN APP
+
+####[Reducers](https://redux.js.org/tutorials/essentials/part-1-overview-concepts#reducers)
+
+A Reducer returns the proper current state with applicable changes as compared to the previous state when it's passed a new state. They work similarly to the Array.reduce method.
+
+####[Redux Store](https://redux.js.org/tutorials/fundamentals/part-4-store#redux-store)
+
+The current live state for your app lives in the Redux Store.
+
+####[Selectors](https://redux.js.org/tutorials/essentials/part-1-overview-concepts#selectors)
+
+A selector returns a piece of the live Redux state as stored in the live Redux store.
+
+####[Dispatch](https://redux.js.org/tutorials/essentials/part-1-overview-concepts#dispatch)
+
+A Dispatch is a method with triggers an action that in turn updates the Redux store.
+
+**NOTE:** The built-in Redux template we used includes a great counter example of how Redux works. You can also take a look at [Redux's explanation of the counter example](https://redux.js.org/introduction/examples/).
+
+## Okta Authentication Setup
+
+For this demo app, we'll be using [Okta's SPA redirect model](https://developer.okta.com/docs/guides/sign-into-spa-redirect/react/main/) to authenticate and fetch user info.
+
+1. First, install the CLI:
```

Comment:
You should be able to remove the instructions to install the CLI. We have a snippet for this you can use. Link includes how-to, configs, and what the snippet renders.

https://github.com/oktadev/okta-blog#use-the-okta-cli-to-register-your-app